### PR TITLE
Schema testing

### LINKED ISSUE #165: Schemas should be tested
**Is your feature request related to a problem? Please describe.**

Our schemas, e.g. `config.schema.json` should be tested with example input JSON.

**Describe the solution you'd like**

One directory with failing cases and one with passing for every schema.

### PR DESCRIPTION
**Pull request by zepatrik**
## Related issue

closes #165 

## Proposed changes

This is a first implementation of our schema "snapshot" tester. It will traverse a success and a failure directory and test the schema against files found in there. It also extracts definitions from that schema to test those individually.

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [x] I have read the [security policy](../security/policy)
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)
- [ ] I have documented my changes in the [developer guide](https://github.com/ory/docs) (if appropriate)

## Further comments

Will maybe end up in it's own package as a stand-alone cli test tool.

### CODE REVIEW

**Comment by zepatrik** [comment]
Data/snapshots/... could also be generated as we "fixed" circular dependencies by not resolving refs and just checking the src of the ref

File: test/schema_tests.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	return nil, errors.Errorf("could not find schema with name %s", n)
+}
+
+func TestSchemas(t *testing.T) {
+	t.Run("test ./config.schema.json", SchemaTestRunner("../docs", "config"))
+}
+
+func SchemaTestRunner(spath string, sname string) func(*testing.T) {
+	return func(t *testing.T) {
+		f, err := os.Open(fmt.Sprintf("%s/%s.schema.json", spath, sname))
```
Comment (aeneasr) [inline test/schema_tests.go]: You can use `ioutil.ReadFile()` here as well

File: test/schema_tests.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+				return true
+			})
+		}
+
+		RunCases(t, schemas, fmt.Sprintf("./%s.schema.test.success", sname), success)
+		RunCases(t, schemas, fmt.Sprintf("./%s.schema.test.failure", sname), failure)
+	}
+}
+
+func RunCases(t *testing.T, ss schemas, dir string, r result) {
```
Comment (aeneasr) [inline test/schema_tests.go]: Rename `r` to `expected` for better readability

File: test/schema_tests.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+		sx, err := gojsonschema.NewSchema(gojsonschema.NewStringLoader(s.raw))
+		if err != nil {
+			return err
+		}
+
+		s.s = sx
+	}
+
+	var l gojsonschema.JSONLoader
+	if strings.HasSuffix(path, "yaml") {
```
[the remark this answers is not in the document]
Comment (zepatrik) [reply test/schema_tests.go]: yes

**Pull request merged**